frontend/src/css/road_details.css

```
@@ -267,3 +267,82 @@ input[type='checkbox'] {
 .image-gallery-page.center-images {
   justify-content: center;
 }
+
+/* Css for click image to zoom */
+.image-click-modal {
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background: rgba(0, 0, 0, 0.5); /* Semi-transparent black background */
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  z-index: 1000; /* Ensure the modal is displayed above other components */
+}
+.image-modal-content {
+  position: relative;
+  max-height: 50vw;
+  max-width: auto;
+  overflow: auto;
+  background-color: white;
+  z-index: 1100; /* Ensure content inside the modal is above the background */
+}
+.close-icon {
+  position: absolute;
+  top: 10px;
+  right: 10px;
+  cursor: pointer;
+}
+
+.image-modal-content img {
+  height: 50vw;
+  width: auto;
+}
```

```suggestion
   display:block
}
```
To make sure that the div have the same size as the image (I found this here: https://stackoverflow.com/questions/21726898/div-with-image-has-a-bigger-height-than-expected)

It would fix the only bug left for me

---

Thank you! I have fixed it according to the link, haha, and sent the pull request again